encfs/RawFileIO.cpp

```diff
@@ -220,14 +221,17 @@ bool RawFileIO::write(const IORequest &req) {
   ssize_t bytes = req.dataLen;
   off_t offset = req.offset;
 
+  int eno = 0;
   while ((bytes != 0) && retrys > 0) {
+    errno = 0;
```

Comment:
think my earlier comments were lost.
I'd rather not mutate errno if not necessary, and I don't think you need to do that here.

ie:
```
while ((bytes != 0 ...) {
  auto writeSize = ::pwrite(...);
  if (writeSize < 0) {
     auto eno = errno;
     ...
     return -eno;
  }
  ...

  knownSize = false;
  return -EIO;
}
```